tests/test_linear_model.py

```diff
@@ -65,12 +136,71 @@ def test_linear_model(self):
         with self.assertRaises(TypeError):
             m.C = True # boolean
             m.matrixCheck()
-        # @E
+        with self.assertRaises(AttributeError):
+            m.C = np.array(2) # 0-D array
+            m.matrixCheck()
+        with self.assertRaises(AttributeError):
+            m.C = np.array([[0, 0], [1, 1]]) # 2-D array
+            m.matrixCheck()
+        with self.assertRaises(AttributeError):
+            m.C = np.array([[1, 0, 2]]) # extra column values per row
+            m.matrixCheck()
+        with self.assertRaises(AttributeError):
+            m.C = np.array([[1]]) # less column values per row
+            m.matrixCheck()
+        with self.assertRaises(AttributeError): 
+            m.C = np.array([[0, 0], [1, 1], [2, 2]]) # extra row
+            m.matrixCheck()
+        with self.assertRaises(AttributeError): 
+            m.C = np.array([[]]) # less row 
+            m.matrixCheck()
+        m.C = np.array([[1, 0]])
+        m.matrixCheck()
+
+
+# Included some tests that are checking if exceptions are being thrown without matrxiCheck being invoked
```

Comment:
Does this comment refer to the code below it? I'm not sure I understand what it refers to. This is a good idea for tests, though!

Reply:
That comment was referring to the tests underneath (for parameter D). Some of the tests for parameter D are checking correct dimensions and type without matrixCheck.

Reply:
Ah, I see it now. Thank you for clarifying!